ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSource.mm

```diff
@@ -260,17 +260,14 @@ static void verifyChangeset(CKTransactionalComponentDataSourceChangeset *changes
                             CKTransactionalComponentDataSourceState *state,
                             NSArray<id<CKTransactionalComponentDataSourceStateModifying>> *pendingAsynchronousModifications)
 {
-#if CK_ASSERTIONS_ENABLED
```

Comment:
I'd appreciate thoughts on this. Determining if the changeset is valid involves some work, though I don't believe the performance characteristics are poor enough to warrant this check.